string.c

```diff
@@ -661,7 +661,7 @@ bool w_string_is_null_terminated(w_string_t *str) {
      w_string_is_null_terminated(str->slice));
 }
 
-bool w_string_strlen(w_string_t *str) {
+size_t w_string_strlen(w_string_t *str) {
```

Comment:
This should be included in the split out from the previous diff. Remember that each diff must stand independently.